trio/_socket.py

```diff
@@ -446,36 +550,72 @@ def __init__(self, sock: _stdlib_socket.socket):
     # Simple + portable methods and attributes
     ################################################################
 
-    # NB this doesn't work because for loops don't create a scope
-    # for _name in [
-    #         ]:
-    #     _meth = getattr(_stdlib_socket.socket, _name)
-    #     @_wraps(_meth, assigned=("__name__", "__doc__"), updated=())
-    #     def _wrapped(self, *args, **kwargs):
-    #         return getattr(self._sock, _meth)(*args, **kwargs)
-    #     locals()[_meth] = _wrapped
-    # del _name, _meth, _wrapped
-
-    _forward = {
-        "detach",
-        "get_inheritable",
-        "set_inheritable",
-        "fileno",
-        "getpeername",
-        "getsockname",
-        "getsockopt",
-        "setsockopt",
-        "listen",
-        "share",
-    }
-
-    def __getattr__(self, name):
-        if name in self._forward:
-            return getattr(self._sock, name)
-        raise AttributeError(name)
-
-    def __dir__(self) -> Iterable[str]:
-        return [*super().__dir__(), *self._forward]
+    # forwarded methods
+    def detach(self) -> int:
+        return self._sock.detach()
+
+    def fileno(self) -> int:
+        return self._sock.fileno()
+
+    def getpeername(self) -> Any:
+        return self._sock.getpeername()
+
+    def getsockname(self) -> Any:
+        return self._sock.getsockname()
+
+    @overload
+    def getsockopt(self, __level: int, __optname: int) -> int:
```

Comment:
We dropped support for 3.7

Reply:
I'm not sure what you're referring to here, I'm not finding what's changed and this is what's in typeshed.

Reply:
I'm not sure what you're referring to here, I'm not finding what's changed and this is what's in typeshed.

Reply:
3.8 introduced `/` pos arguments, so we can use that instead of the `__arg` hack.

Reply:
Changed it in a bunch of places, but I failed to remove them from `sendto`, and from the ones where `_make_simple_sock_method_wrapper` is used to verify the signature.